plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -0,0 +1,130 @@
+                if (a.type.toString().equals("int")) {
+                    type = "int";
+                } else if (a.type.toString().equals("class java.lang.String")) {
+                    type = "string";
+                } else if (a.type.toString().equals("boolean")) {
+                    type = "boolean";
+                } else {
+                    type = "object";
+                }
+                schemaString.append(a.name + ": " + "{\n" +
+                        "      \"type\": \"" + type + "\",\n" +
+                        "      \"$ref\": \"#/definitions/" + a.type.getName() + "\"\n" +
+                        "    },");
+            }
+        }
+//        System.out.println(schemaString);
+
+        /**
+         * Used to generate the schema for the descriptors
+         * Finds out the instance of each of the configurators
+         * and gets the required descriptors from the instance method.
+         */
+        ConfigurationAsCode configurationAsCode = ConfigurationAsCode.get();
+        configurationAsCode.getConfigurators()
+                .stream()
+                .forEach(configurator -> {
+
+                    DescriptorExtensionList descriptorExtensionList = null;
+
+                    if(configurator instanceof HeteroDescribableConfigurator) {
+                        HeteroDescribableConfigurator heteroDescribableConfigurator = (HeteroDescribableConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(heteroDescribableConfigurator.getTarget());
+
+                    } else if (configurator instanceof DataBoundConfigurator){
+                        DataBoundConfigurator dataBoundConfigurator = (DataBoundConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(dataBoundConfigurator.getTarget());
+
+
+                    }  else if (configurator instanceof JenkinsConfigurator) {
+                        JenkinsConfigurator jenkinsConfigurator = (JenkinsConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(jenkinsConfigurator.getTarget());
+
+                    }  else if (configurator instanceof GlobalConfigurationCategoryConfigurator) {
+                        GlobalConfigurationCategoryConfigurator globalConfigurationCategoryConfigurator = (GlobalConfigurationCategoryConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(globalConfigurationCategoryConfigurator.getTarget());
+
+                    }  else if (configurator instanceof UnsecuredAuthorizationStrategyConfigurator) {
+                        UnsecuredAuthorizationStrategyConfigurator unsecuredAuthorizationStrategyConfigurator = (UnsecuredAuthorizationStrategyConfigurator) configurator;
+                        descriptorExtensionList = j.jenkins.getDescriptorList(unsecuredAuthorizationStrategyConfigurator.getTarget());
```

Comment:
Same problem here as well.

Reply:
From looking at the code I have a suspicion that this should only be done at the attribute level but need to add some debug to see exactly what schema file is called when

Reply:
Yeah we might be able to grab the attributes, but I am not sure if its what we are looking for.